torch/cuda/__init__.py

```diff
@@ -648,26 +648,28 @@ def _parse_visible_devices() -> Union[List[int], List[str]]:
         hip_devices = os.getenv("HIP_VISIBLE_DEVICES")
         rocr_devices = os.getenv("ROCR_VISIBLE_DEVICES")
 
+        # You must take care if both HIP and ROCR env vars are set as they have
+        # different meanings. ROCR only accepts a list of ints which then reduces
+        # the number of GPUs that HIP can select from. The HIP env var can accept
+        # either a list of ints or a list of UUIDs prefixed with 'GPU-'.
         if rocr_devices is not None:
-            # Mostly required for ROCm to make sure ROCr visible devices
-            # is respected, this ensures we do not return a list of devices
-            # that exceeds the total available supplied via ROCR_VISIBLE_DEVICES
-            var = rocr_devices
-
-        if hip_devices is not None:
-            # If ROCr devices have been set, the hip visible devices would
-            # be a subset of those. HIP_VISIBLE_DEVICES can only contain
-            # integer indices so we can use the ROCr visible devices as a key
-            if rocr_devices is not None:
-                hip_device_list = [int(dev) for dev in hip_devices.split(",")]
-                rocr_device_list = rocr_devices.split(",")
-                var = ",".join(
-                    rocr_device_list[dev]
-                    for dev in hip_device_list
-                    if dev in rocr_device_list
-                )
-            else:
+            rocr_count = len(rocr_devics.split(","))
+            if hip_devices is not None:
+                if hip_devices.startswith("GPU-"):
```

Comment:
could HIP_VISIBLE_DEVICES be set to `0,GPU-uuid,2`? If so, this check should be on `any(dev for dev in hip_devices.split(","))`

Reply:
@jeffdaily Although not a common use case it could be possible for the user to use  ROCR_VISIBLE_DEVICES in combination with HIP_VISIBLE_DEVICES with UUIDs

e.g. may use ROCR to expose only GPUs "0,1" in some environment but the user in that environment may want to use UUIDs in HIP_VISIBLE_DEVICES to ensure they run on the requested gpu.

Reply:
I was wrong about ROCR env var only taking ints.  It can take UUIDs also.  So ROCR and HIP env vars are similar today, but ROCR env var activates first lower-level than HIP env var so it will reduce the number of devices that HIP can select from.  I think the best we can do now is go back to the simple check that HIP env var isn't asking for more devices than are visible.